src/core/lib/gprpp/ref_counted.h

```diff
@@ -69,7 +69,14 @@ class RefCount {
   // Note: RefCount tracing is only enabled on debug builds, even when a
   //       TraceFlag is used.
   template <typename TraceFlagT = TraceFlag>
-  constexpr explicit RefCount(Value init = 1, TraceFlagT* trace_flag = nullptr)
+  constexpr explicit RefCount(
+      Value init = 1,
+      TraceFlagT*
+#ifndef NDEBUG
```

Comment:
Why leave it unnamed only for debug scenario?

Reply:
It's named only for the debug scenario (`#ifndef NDEBUG`) because that's the only place where it gets used for logging/etc.